esp-config/src/generate/mod.rs

```diff
@@ -45,8 +46,466 @@ impl fmt::Display for Error {
+
+        eval_ctx
+            .set_function(
+                "ignore_feature_gates".into(),
+                evalexpr::Function::<evalexpr::DefaultNumericTypes>::new(move |arg| {
+                    if let evalexpr::Value::Empty = arg {
+                        Ok(evalexpr::Value::Boolean(ignore_feature_gates))
+                    } else {
+                        Err(evalexpr::EvalexprError::CustomMessage(format!(
+                            "Bad argument: {:?}",
+                            arg
+                        )))
+                    }
+                }),
+            )
+            .map_err(|err| Error::Parse(err.to_string()))?;
+    }
+    for option in config.options.clone() {
+        let active = evalexpr::eval_with_context(&option.active, &eval_ctx)
+            .map_err(|err| {
+                Error::Parse(format!(
+                    "Error evaluating '{}', error = {:?}",
+                    option.active, err
+                ))
+            })?
+            .as_boolean()
+            .map_err(|err| {
+                Error::Parse(format!(
+                    "Error evaluating '{}', error = {:?}",
+                    option.active, err
+                ))
+            })?;
+
+        let constraint = {
+            let mut active_constraint = None;
+            if let Some(constraints) = &option.constraints {
+                for constraint in constraints {
+                    if evalexpr::eval_with_context(&constraint.if_, &eval_ctx)
+                        .map_err(|err| {
+                            Error::Parse(format!(
+                                "Error evaluating '{}', error = {:?}",
+                                constraint.if_, err
+                            ))
+                        })?
+                        .as_boolean()
+                        .map_err(|err| {
+                            Error::Parse(format!(
+                                "Error evaluating '{}', error = {:?}",
+                                constraint.if_, err
+                            ))
+                        })?
+                    {
+                        active_constraint = Some(constraint.type_.clone());
+                        break;
+                    }
+                }
+            };
+
+            if option.constraints.is_some() && active_constraint.is_none() {
+                panic!(
+                    "No constraint active for crate {}, option {}",
+                    config.krate, option.name
+                );
+            }
+
+            active_constraint
+        };
+
+        let default_value = {
+            let mut default_value = None;
+            for value in option.default.clone() {
+                if evalexpr::eval_with_context(&value.if_, &eval_ctx)
+                    .map_err(|err| {
+                        Error::Parse(format!(
+                            "Error evaluating '{}', error = {:?}",
+                            value.if_, err
```

Comment:
Nightly clippy will be mad.
```suggestion
                            "Error evaluating '{}', error = {err:?}",
                            value.if_
```